command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -79,6 +114,138 @@ func (c *cmd) Run(args []string) int {
 	return 0
 }
 
+type typeStats struct {
+	Name       string
+	Sum, Count int
+}
+
+// countingReader helps keep track of the bytes we have read
+// when reading snapshots
+type countingReader struct {
+	wrappedReader io.Reader
+	read          int
+}
+
+func (r *countingReader) Read(p []byte) (n int, err error) {
+	n, err = r.wrappedReader.Read(p)
+	if err == nil {
+		r.read += n
+	}
+	return n, err
+}
+
+// enhance utilizes ReadSnapshot to populate the struct with
+// all of the snapshot's itemized data
+func enhance(file io.Reader) (map[structs.MessageType]typeStats, int, error) {
+	stats := make(map[structs.MessageType]typeStats)
+	var offset int
+	cr := &countingReader{wrappedReader: file}
+	offset = 0
+	handler := func(header *fsm.SnapshotHeader, msg structs.MessageType, dec *codec.Decoder) error {
+		name := structs.MessageType.String(msg)
+		s := stats[msg]
+		if s.Name == "" {
+			s.Name = name
+		}
+		var val interface{}
+		err := dec.Decode(&val)
+		if err != nil {
+			return fmt.Errorf("failed to decode msg type %v, error %v", name, err)
+		}
+
+		size := cr.read - offset
+		s.Sum += size
+		s.Count++
+		offset = cr.read
+		stats[msg] = s
+		return nil
+	}
+	if err := fsm.ReadSnapshot(cr, handler); err != nil {
+		return nil, 0, err
+	}
+	return stats, offset, nil
+
+}
+
+// readStats takes the information generated from enhance and creates human
+// readable output from it
+func (c *cmd) readStats(stats map[structs.MessageType]typeStats, offset int) (bytes.Buffer, error) {
+	// Output stats in size-order
+	ss := make([]typeStats, 0, len(stats))
+
+	for _, s := range stats {
+		ss = append(ss, s)
+	}
+
+	// Sort the stat slice
+	sort.Slice(ss, func(i, j int) bool { return ss[i].Sum > ss[j].Sum })
+
+	var b bytes.Buffer
+
+	tw := tabwriter.NewWriter(&b, 8, 8, 6, ' ', 0)
+	fmt.Fprintln(tw, "\n Type\tCount\tSize\t")
+	fmt.Fprintf(tw, " %s\t%s\t%s\t", "----", "----", "----")
+	// For each different type generate new output
+	for _, s := range ss {
+		fmt.Fprintf(tw, "\n %s\t%d\t%s\t", s.Name, s.Count, ByteSize(uint64(s.Sum)))
+	}
+	fmt.Fprintf(tw, "\n %s\t%s\t%s\t", "----", "----", "----")
+	fmt.Fprintln(tw, "\n Total\t\t", ByteSize(uint64(offset)))
+
+	if err := tw.Flush(); err != nil {
+		c.UI.Error(fmt.Sprintf("Error rendering snapshot info: %s", err))
+		return b, err
+	}
+
+	return b, nil
+
+}
+
+// ByteSize returns a human-readable byte string of the form 10M, 12.5K, and so forth.  The following units are available:
+//	T: Terabyte
+//	G: Gigabyte
+//	M: Megabyte
+//	K: Kilobyte
+//	B: Byte
+// The unit that results in the smallest number greater than or equal to 1 is always chosen.
```

Comment:
This comment should match the formatting below:
```suggestion
// ByteSize returns a human-readable byte string of the form 10MB, 12.5KB, and so forth.  The following units are available:
//	TB: Terabyte
//	GB: Gigabyte
//	MB: Megabyte
//	KB: Kilobyte
//	B: Byte
// The unit that results in the smallest number greater than or equal to 1 is always chosen.
```